Block Editor: Add documentation for SpacingSizesControl component

Infinite-Null opened the pull request:
Related to: #22891
Closes: #52070

## What?
Adds documentation for the SpacingSizesControl component by:

- Adding a comprehensive README.md file
- Adding JSDoc comments with proper syntax, examples, and prop descriptions

## Why?
The SpacingSizesControl component currently lacks documentation, making it difficult for developers to understand and implement the component correctly:

- Clear documentation of all available props and their usage
- Examples showing how to implement the component properly
- Documentation of component views and internal structure

The documentation follows established patterns from other components while providing specific details relevant to SpacingSizesControl, making it easier for developers to use this component effectively in their blocks.

Infinite-Null committed:
Block Editor: Add documentation for SpacingSizesControl component

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/README.md:
```diff
@@ -0,0 +1,113 @@
+	return (
+		<View className="tools-panel-item-spacing">
+			<SpacingSizesControl
+				values={ { all: value } }
```
I don’t see where this `all` key is supported. I got here from testing #68339 so I even tried this with that story and it didn’t cause errors but also didn’t seem to do anything. It looks like this ought to be one of `top`, `right`, `bottom`, `left`. It’s probably best to use them all just for a complete example even though its seems they are all optional.

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/README.md:
```diff
@@ -0,0 +1,113 @@
+		defaultValues: { px: 100, em: 10, rem: 10, vw: 10, vh: 25 },
+	} );
+
+	const handleOnChange = ( unprocessedValue ) => {
+		onChange( unprocessedValue.all );
+	};
```
I think it’d be fine to not have the wrapper and simply pass `onChange` onto `SpacingSizesControl`. At the least, the `unprocessedValue.all` seems like it shouldn’t be there (per my other comment about `all` not being a supported part of the API).

On the other hand, if it seems nice to demonstrate what the received value will consist of it could be something like this:
```suggestion
	const handleOnChange = ( { top, right = '0px', bottom, left = '0px' } ) => {
		onChange( { top, right, bottom, left } );
	};
```

Infinite-Null committed:
Refactor `SpacingSizesControl` to use individual side values

Infinite-Null commented:
Hi @stokesman,
Thank you for the feedback!
I have implemented the requested changes. Please review them at your convenience.

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/index.js:
````diff
@@ -25,6 +25,50 @@ import {
+ *
+ * @example
+ * ```jsx
+ * function SpacerBlock() {
+ *   return (
+ *     <View className="tools-panel-item-spacing">
+ *       <SpacingSizesControl
+ *         values={ {
+ *				top: '0px',
+ *				right: '0px',
+ *				bottom: '0px',
+ *				left: '0px',
+ *			} }
+ *         onChange={handleOnChange}
+ *         label="Height"
+ *         units={units}
+ *         allowReset={false}
+ *         splitOnAxis={false}
+ *         showSideInLabel={false}
+ *       />
+ *     </View>
+ *   );
+ * }
````
Going off [examples in the components package](https://github.com/search?q=repo%3AWordPress%2Fgutenberg+%60%60%60jsx++path%3A%2F%5Epackages%5C%2Fcomponents%5C%2Fsrc%5C%2F%28%5Cw%7C-%29*%5C%2F.*%5C.%28js%7Cts%29x%3F%2F&type=code&p=1) I think this should be more minimal yet also more complete. I suggest to:
- Remove `units` since it isn’t a prop
- Remove `allowReset`, `splitOnAxis`, `showSideInLabel` since they are optional props
- Remove the `<View>` wrapper
- Add the import for `SpacingSizesControl`
- Add an import for `useState` and usage in the component:
  ```
   *   const [ sides, setSides ] = useState( {
   *     top: '0px',
   *     right: '0px',
   *     bottom: '0px',
   *     left: '0px',
   *   } );
   ```
- Change the `values` prop to `values={ sides }`
- Change the `onChange` prop to `onChange={ setSides }`
- Rename `SpacerBlock` to `Example`
- Change the `label` prop to `label="Sides"`

Also the example in the README should be the essentially the same (apart from formatting).

Infinite-Null replied:
Hi @stokesman,
I have made the necessary changes please have a look at it at your convenience.
Thank You!

stokesman reviewed (commented):
Thanks for the update 🙇. I had a look at prior art around the codebase and have some more suggestions.

Infinite-Null committed:
Simplify component example and documentation
- Removed unnecessary props and wrappers
- Added proper state management with useState hook
- Ensured consistent documentation between the component and README
- Replaced SpacerBlock with Example component
- Added proper imports for SpacingSizesControl and useState
- Removed unnecessary View wrapper and optional props
- Updated README to match the new implementation

Infinite-Null committed:
Merge branch 'trunk' into docs/add-spacing-sizes-control-readme-js-doc

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/README.md:
```diff
@@ -0,0 +1,99 @@
+2. Axial: Control horizontal and vertical sides together
+3. Custom: Control each side separately
+
+![Spacing Sizes Control](https://i.postimg.cc/3RkzzfL6/Screenshot-2025-01-10-at-8-07-55-AM.png)
```
I think this should be omitted. Images are nice but liable to become outdated. Besides that it seems all images should be sourced from wordpress.org.

stokesman replied:
I think this should be omitted. Images are nice but liable to become outdated. Besides that it seems all images should be sourced from wordpress.org.

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/README.md:
```diff
@@ -0,0 +1,99 @@
+# Spacing Sizes Control
+
+The SpacingSizesControl component provides a user interface for controlling spacing values in blocks. It supports single, axial, and separated input controls for different spacing configurations.
+
+## Description
+
+The SpacingSizesControl component is a flexible control that allows users to modify spacing values for different sides of a block. It supports three viewing modes:
+
+1. Single: Control one side at a time
+2. Axial: Control horizontal and vertical sides together
+3. Custom: Control each side separately
```
I think this should be consolidated. I didn’t see other READMEs with a `## Description` so that can be removed. Then the text before it and after it is redundant so keep only the before or after.

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/README.md:
````diff
@@ -0,0 +1,99 @@
+## Usage
+
+```jsx
+import { SpacingSizesControl } from '@wordpress/block-editor';
````
```suggestion
import { __experimentalSpacingSizesControl as SpacingSizesControl } from '@wordpress/block-editor';
```

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/index.js:
````diff
@@ -25,6 +25,49 @@ import {
+ *
+ * @example
+ * ```jsx
+ * import { SpacingSizesControl } from '@wordpress/block-editor';
````
```suggestion
 * import { __experimentalSpacingSizesControl as SpacingSizesControl } from '@wordpress/block-editor';
```

stokesman commented on packages/block-editor/src/components/spacing-sizes-control/README.md:
```diff
@@ -0,0 +1,99 @@
+
+-   Type: `String`
+-   Required: Yes
+-   Description: Label for the control (e.g., "Height").
```
I think the example `(e.g., "Height")` is not a great one. I don’t think we need such an example anyway so I’d just remove that bit.

stokesman reviewed (commented):
Thanks for the updates again 🙇.

Infinite-Null committed:
Merge branch 'trunk' into docs/add-spacing-sizes-control-readme-js-doc

Infinite-Null committed:
Consolidate README and update import examples

- Removed redundant description section in README
- Consolidated component description into a single paragraph
- Updated import examples to use `__experimentalSpacingSizesControl` alias
- Removed screenshot reference
- Standardized prop descriptions

Infinite-Null commented:
Hi @stokesman,
Thank you for the follow-up feedback!
I have implemented the necessary changes. Please review them at your convenience.

stokesman reviewed (approved):
This seems adequate to me. Thank you @Infinite-Null!
